File: app/src/main/java/top/atmb/autumnbox/acp/processor/ACPDataBuilder.java
package top.atmb.autumnbox.acp.processor;

import top.atmb.autumnbox.acp.ACP;

/**
 * Created by zsh24 on 01/21/2018.
 */

public final class ACPDataBuilder {
    private static final String TAG = "ACPDataBuilder";
    private byte fCode = ACP.FCODE_UNKNOW_ERR;
    private byte[] data = new byte[0];
    public int getTotalLenght(){
        return getDataLenght() + 1;
    }
    public int getDataLenght(){
        return data.length;
    }
    public byte fCode(){
        return fCode;
    }
    public void setfCode(byte fCode){
        this.fCode = fCode;
    }
    public void setData(byte[] bytes){
        ACP.printLog(TAG,"set data" + bytes.length);
        this.data = bytes;
    }
    public byte[] toBytes(){
        byte[] result = new byte[getTotalLenght()];
        result[0] = fCode;
        if(getDataLenght() > 0){
            System.arraycopy(data,0,result, 1,data.length);
        }
//        ACP.printLog(TAG,"fCode->"+ result[0]);
//        ACP.printLog(TAG,"lenght ->" + getTotalLenght());
        return result;
    }
}

File: app/src/main/java/top/atmb/autumnbox/acp/ACPDataBuilder.kt
package top.atmb.autumnbox.acp

import java.util.*

/**
 * Created by zsh24 on 01/22/2018.
 */
data class ACPDataBuilder(var fCode:Byte = 1,var data:ByteArray = ByteArray(0)){
    val totalSize:Int get() { return data.size + 1 }
    val dataSize:Int get() {return data.size}

    fun toBytes():ByteArray{
        var result:Array<Byte> = ByteArray(totalSize).toTypedArray()
        result[0] = fCode
        if(data.isNotEmpty()){
            System.arraycopy(data.toTypedArray(), 0,result,1,dataSize)
        }
        return result.toByteArray()
    }
}